src/OpenTelemetry/Metrics/ExponentialBuckets.cs

```diff
@@ -0,0 +1,72 @@
+// <copyright file="ExponentialBuckets.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Metrics;
+
+internal class ExponentialBuckets
+{
+    private readonly CircularBufferBuckets buckets;
+
+    internal ExponentialBuckets(CircularBufferBuckets buckets)
+    {
+        this.buckets = buckets;
+    }
+
+    public Enumerator GetEnumerator() => new(this.buckets);
+
+    /// <summary>
+    /// Enumerates the bucket counts of an exponential histogram.
+    /// </summary>
+    // Note: Does not implement IEnumerator<> to prevent accidental boxing.
+    public struct Enumerator
+    {
+        private readonly int lastIndex;
+        private readonly CircularBufferBuckets buckets;
+        private int index;
+
+        internal Enumerator(CircularBufferBuckets buckets)
+        {
+            this.buckets = buckets;
+            this.index = buckets.Offset;
+            this.Current = default;
+            this.lastIndex = buckets.Size - 1;
+        }
+
+        /// <summary>
+        /// Gets the bucket count at the current position of the enumerator.
+        /// </summary>
+        public long Current { get; private set; }
```

Comment:
How do we retrieve the actual bucket boundary that this count belongs to?

Reply:
It's probably a good idea to add a property to provide `Base` in `ExponentialBucketSnapshot`.

Reply:
Not sure it'll be necessary. `ExponentialBucketSnapshot` is basically what exporters get in order to translate stuff to their wire format. As seen in the proto, there's no need to actually compute bucket boundaries on behalf of the exporter. The back end can compute bucket boundaries simply with scale, offset, and the array of bucket counts.

In fact, one of the benefits of exponential histograms is that for a histogram of the roughly the same size, an exponential histogram requires sending less data over the wire than an explicit bounds histogram. Because you don't send the boundaries.